src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,132 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.*;
+
+@Embeddable
+public class Sections {
+
+    private static final int MIN_SECTIONS_SIZE = 1;
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    protected Sections() {
+    }
+
+    public void add(Section section) {
+        validateAddable(section);
+        if (isEmpty()) {
+            sections.add(section);
+            return;
+        }
+        divideByUpStation(section);
+        divideByDownStation(section);
+        sections.add(section);
+    }
+
+    private void validateAddable(Section section) {
+        if (contains(section.getUpStation()) && contains(section.getDownStation())) {
+            throw new IllegalArgumentException("이미 등록된 구간 입니다.");
+        }
+        if (!isEmpty() && !contains(section.getUpStation()) && !contains(section.getDownStation())) {
+            throw new IllegalArgumentException("구간 추가를 위해서는 기존 구간과의 연결점이 필요합니다. 역 정보를 확인해주세요.");
+        }
+    }
+
+    private void divideByUpStation(Section section) {
+        sections.stream()
+                .filter(it -> it.getUpStation().equals(section.getUpStation()))
+                .findFirst()
+                .ifPresent(it -> it.updateUpStation(section.getDownStation(), section.getDistance()));
+    }
+
+    private void divideByDownStation(Section section) {
+        sections.stream()
+                .filter(it -> it.getDownStation().equals(section.getDownStation()))
+                .findFirst()
+                .ifPresent(it -> it.updateDownStation(section.getUpStation(), section.getDistance()));
+    }
+
+    public void remove(Line line, Station station) {
+        validateRemovable(station);
+
+        Optional<Section> upSection = findUpSection(station);
+        Optional<Section> downSection = findDownSection(station);
+
+        if (upSection.isPresent() && downSection.isPresent()) {
+            addMergeSections(line, upSection, downSection);
+        }
+
+        upSection.ifPresent(it -> sections.remove(it));
+        downSection.ifPresent(it -> sections.remove(it));
+    }
+
+    private void validateRemovable(Station station) {
+        if (sections.size() <= MIN_SECTIONS_SIZE) {
+            throw new IllegalStateException("구간의 갯수가 " + sections.size() + " 입니다. 삭제할 수 없습니다.");
+        }
+        if (!contains(station)) {
+            throw new IllegalArgumentException("구간에 존재하지 않는 역입니다. 삭제하고자 하는 역 정보를 확인해 주세요.");
+        }
+    }
+
+    private void addMergeSections(Line line, Optional<Section> upSection, Optional<Section> downSection) {
```

Comment:
이 경우는 isPresent를 upSection, downSection이 존재 하는 부분이 검증되었으므로, Section을 파라미터로 받아도 괜찮지 않을까요?

```suggestion
    private void addMergeSections(Line line, Section upSection, Section downSection) {
```

Reply:
헛 간단히 생각할 수 있는 내용이었는데, 제가 너무 Optional 클래스에 집착했던 것 같네요😨
확인 감사합니다~! 수정해서 반영하였습니다!